components/script/dom/websocket.rs

```diff
@@ -299,17 +299,7 @@ impl WebSocket {
         let chan = global.r().networking_task_source();
         let address = Trusted::new(self, chan.clone());
 
-        let new_buffer_amount = (self.buffered_amount.get() as u64) + data_byte_len;
-        if new_buffer_amount > (u32::max_value() as u64) {
-            self.buffered_amount.set(u32::max_value());
-            self.full.set(true);
-
-            let _ = self.Close(None, None);
-            return Ok(false);
-
-        }
-
-        self.buffered_amount.set(new_buffer_amount as u32);
+        self.buffered_amount.set(self.buffered_amount.get() + data_byte_len);
```

Comment:
https://doc.rust-lang.org/nightly/std/primitive.u64.html#method.checked_add is the usual way to check for overflow in a situation like this.